### PR TITLE
Updated README.md and added LoRA finetuning code 

### PR DESCRIPTION
**Pull request by Elsword016**
Hello! 
I have updated the main README.md and added details where necessary. Also:

- Added a folder within the `counting_intersections` folder containing a notebook for fine-tuning the `google/paligemma-3b-pt-448` on the intersection dataset with LoRA adapters without freezing any model blocks.  
- Tensorboard log for that run.
- README: Detailing the training arguments used, how to access the trained checkpoint as well as the a link to run the notebook in Google Colab.

I have no compute left to improve the model :(

### CODE REVIEW

**Comment by ariG23498** [comment]
Really cool project. I have not reviewed it completely, but here are my suggestions:

1. Let's not add another folder under `count_intersection`. It would be more than okay to have a different file in the same directory with the proper name.
2. It would be great if we could do `.py` scripts for training and not `.ipynb` files. That way it would be easier for me to review the bits.
3. Could we switch to `wandb` for logging? I am not against tensorborad, but I don't want the logging runs to stay inside the github repo.
4. The dataset is of `224x224` resolution, I think we should use the model that supports it.

I am looking forward to this work. Also I will be OOO from 13th to 16th. If you do not get a review at that time, please do not think that I am not interested in the project. Looking forward to where we can take this.

**Comment by Elsword016** [comment]
Will make the changes 🫡

**Comment by ariG23498** [comment]
Let me know once you would like my review on this.

**Comment by ariG23498** [comment]
Hey @Elsword016 any updates on this?

**Comment by Elsword016** [comment]
Hi! I will update it during the weekend. Sorry for the delay, I was in a conference for the week

**Comment by ariG23498** [comment]
No worries. Hope your conference went well 👌🏻

**Comment by Elsword016** [comment]
Hello! @ariG23498 . I have made the changes as requested.
- Removal of the extra folders
- Putting the fine-tuning code in a single Python script.

**Comment by ariG23498** [comment]
Can we please resolve the merge conflicts first before review?

**Comment by Elsword016** [comment]
I resolve the conflicts, please have a look and let me know, if I need to make further changes

File: .github/README.md
Comment (ariG23498) [inline .github/README.md:1]: Let's keep the README as it [was](https://github.com/ariG23498/fine-tune-paligemma) and not make any changes.

File: count_intersection/README.md [hunk 1]
```diff
@@ -66,8 +66,20 @@ python finetune.py \
   - `push_to_hub`: Whether to upload the fine-tuned model to the Hugging Face Hub.
   - `model_id`: ID used to upload model to the Hub, e.g. `oliveirabruno01/count_intersection-ft-paligemma2-3b-pt-224`. Required if `push_to_hub` is `True`.
 
+Fine-tune the whole model(without freezing any layers) with LoRA:
```
Comment (ariG23498) [inline count_intersection/README.md]:
```suggestion
Fine-tune the whole model (without freezing any layers) with LoRA:
```

File: count_intersection/lora_finetune.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# Initialize trainer
+trainer = Trainer(
+    model=model,
+    train_dataset=train_ds,
+    data_collator=collate_fn,
+    args=training_args
+)
+
+# Start training
+trainer.train()
```
Comment (ariG23498) [inline count_intersection/lora_finetune.py:99]: Would be great if you could also add the model push to hub after train.

Wdyt?

**Review by ariG23498 — CHANGES_REQUESTED** [review]
Minor nits. After that we can merge.

File: count_intersection/lora_finetune.py (hunk 2 above)
Comment (Elsword016) [reply count_intersection/lora_finetune.py:99]: Okay I have added the `trainer.push_to_hub()` after training.

**Review by ariG23498 — APPROVED** [review]
Thank you for your contribution!

Merging.

**Pull request merged**